airflow/jobs/local_task_job_runner.py

```diff
@@ -303,6 +306,14 @@ def heartbeat_callback(self, session: Session = NEW_SESSION) -> None:
                 if dagrun_timeout and execution_time > dagrun_timeout:
                     self.log.warning("DagRun timed out after %s.", execution_time)
 
+            # If process still runs after being marked as success, let it run until configured overtime
+            # if there are configured listeners
+            if (
+                ti.state == TaskInstanceState.SUCCESS
```

Comment:
A naïve question, but can the scenario you described occur if a task is marked as failed as well?

Reply:
In this case, it's actually ran before state is formally changed in DB: 
https://github.com/apache/airflow/blob/7dc2b5243621f7a15ed1b44e1bf6eb81670f68d2/airflow/models/taskinstance.py#L3130

Reply:
Maybe it's worth mentioning it somehow in the yaml ? Right now we have:
>     task_listener_timeout:
>       description: |
>        Maximum possible time (in seconds) that task listener will have for their execution.

I know config is not the place to explain how airflow works, but after reading just that i had the impression that this timeout will work for all listener task related methods, not only success one.